projects/bp-gallery/src/components/provider/GrowthBookProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import {
+  GrowthBook,
+  GrowthBookProvider as _GrowthBookProvider,
+} from '@growthbook/growthbook-react';
+import { PropsWithChildren, useEffect } from 'react';
+
+const growthbookApiHost = import.meta.env.VITE_REACT_APP_GROWTHBOOK_APIHOST;
+const growthbookClientKey = import.meta.env.VITE_REACT_APP_GROWTHBOOK_CLIENTKEY;
+
+const growthbook =
+  growthbookApiHost && growthbookClientKey
+    ? new GrowthBook({
+        apiHost: growthbookApiHost,
+        clientKey: growthbookClientKey,
+        enableDevMode: import.meta.env.MODE === 'development',
+        trackingCallback: (experiment, result) => {
+          const w: any = window;
+          const _paq: Array<any> = (w._paq = w._paq || []);
+          _paq.push([
+            'trackEvent',
+            'ExperimentViewed',
+            experiment.key,
+            'v' + String(result.variationId),
+          ]);
+        },
+        onFeatureUsage: (featureKey, result) => {
+          const w: any = window;
+          const _paq: Array<any> = (w._paq = w._paq || []);
+          _paq.push(['trackEvent', 'ExperimentViewed', featureKey, 'v' + String(result)]);
+        },
+      })
+    : undefined;
+
+export const GrowthBookProvider = ({ children }: PropsWithChildren<{}>) => {
+  useEffect(() => {
+    growthbook?.loadFeatures({ autoRefresh: true });
+  }, []);
+  useEffect(() => {
+    // Set user attributes for targeting (from cookie, auth system, etc.)
+    let visitor_id: string;
+    const w: any = window;
+    const _paq: Array<any> = (w._paq = w._paq || []);
+    _paq.push([
+      function (this: any) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+        visitor_id = this.getVisitorId();
+        growthbook?.setAttributes({ ...growthbook.getAttributes(), id: visitor_id });
+      },
+    ]);
+
+    const refresh = (): NodeJS.Timeout | undefined => {
+      if (!growthbook) return;
+      growthbook.refreshFeatures();
+      return setTimeout(refresh, 5000);
```

Comment:
Is there a reason not to use [`setInterval`](https://developer.mozilla.org/en-US/docs/Web/API/setInterval) here? Also, you should probably use `window.setInterval` (or `window.setTimeout`) here to avoid the misleading `NodeJS` types.

Reply:
Used setTimeout because of: 
https://stackoverflow.com/questions/8682622/using-setinterval-to-do-simplistic-continuous-polling

Reply:
Used setTimeout because of [this thread](https://stackoverflow.com/questions/8682622/using-setinterval-to-do-simplistic-continuous-polling)

Reply:
Ah ok makes sense. Then I'd say keep it, but use the `window.` prefix :)